toolchain/check/handle_class.cpp

```diff
@@ -512,6 +512,15 @@ auto HandleParseNode(Context& context, Parse::BaseDeclId node_id) -> bool {
     return true;
   }
 
+  if (!context.struct_type_fields_stack().PeekArray().empty()) {
+    CARBON_DIAGNOSTIC(BaseDeclAfterFieldDecl, Error,
+                      "`{0}` declaration appears after field declaration(s)",
+                      Lex::TokenKind);
+    context.emitter().Emit(node_id, BaseDeclAfterFieldDecl,
+                           Lex::TokenKind::Base);
```

Comment:
Does it work to add a note for the first field location? i.e., something like:

```suggestion
    CARBON_DIAGNOSTIC(FirstFieldDecl, Note,
                      "First field declared here");
    context.emitter().Build(node_id, BaseDeclAfterFieldDecl,
                           Lex::TokenKind::Base).Note(fields.front(), FirstFieldDecl).Emit();
```

(I think `FirstFieldDecl` is a bad name, but I'm struggling with better naming)

Reply:
For now the StructTypeFields don't have source locations for this extra note, but once we track the FieldDecls separately in https://github.com/carbon-language/carbon-lang/pull/4514  I can revisit this to add the note. I'll add a TODO for this.